### ISSUE TITLE
How to run inference  of a single audio file with by your release trained esc_50 model?

### ISSUE THREAD

**Issue by joewale** [RetroCirce/HTS-Audio-Transformer#11]
Hi，RetroCirce

        通过下载您已经训练好的esc_50模型，如何快速通过这个模型来验证输入音频的分类结果？

**Comment by RetroCirce** [comment]
可以查看sed_model.py中对于esc数据集计算acc的代码，其实和audioset相似，就是输出model的结果然后计算，如果你需要得到分类结果，你可以用main.py中的test函数，得到结果直接进行查看

**Comment by joewale** [comment]
Hi，RetroCirce

       这是我写的demo，测试ESC-50单个音频demo，发现结果输出的类别不是预期想要的，而且结果还是随机的，请问这是哪里写错了吗？求帮忙指导，感谢！

`
import os
import json
import torch
import librosa
from model.htsat import HTSAT_Swin_Transformer
import config
import numpy as np


class Audio_Classification:
    def __init__(self, model_path, config):
        super().__init__()

        self.device = torch.device('cuda')
        self.sed_model = HTSAT_Swin_Transformer(
            spec_size=config.htsat_spec_size,
            patch_size=config.htsat_patch_size,
            in_chans=1,
            num_classes=config.classes_num,
            window_size=config.htsat_window_size,
            config = config,
            depths = config.htsat_depth,
            embed_dim = config.htsat_dim,
            patch_stride=config.htsat_stride,
            num_heads=config.htsat_num_head
        )
        ckpt = torch.load(model_path, map_location="cpu")
        self.sed_model.load_state_dict(ckpt['state_dict'], strict=False)
        self.sed_model.eval().to(self.device)


    def predict(self, audiofile):

        if audiofile:
            waveform, sr = librosa.load(audiofile, sr=32000)

            with torch.no_grad():

                x = torch.from_numpy(waveform).float().to(self.device)
                output_dict = self.sed_model(x[None, :], None, True)

                pred = output_dict['clipwise_output']

                pred_post = pred[0].detach().cpu().numpy()

                pred_label = np.argmax(pred_post)

                pred_prob = np.max(pred_post)

            return pred_label, pred_prob



Audiocls = Audio_Classification('exp/HTSAT_ESC_exp=1_fold=1_acc=0.985.ckpt', config)

pred_label, pred_prob = Audiocls.predict("1-28135-A-11.wav")

print('Audiocls predict output: ', pred_label, pred_prob)
`

**Comment by RetroCirce** [comment]
你的config.py方便给我看看嘛

**Comment by joewale** [comment]
这是我的config文件
[Uploading config.txt…]()

**Comment by RetroCirce** [comment]
因为ckpt的key不太一样，你直接使用这个model的时候，前缀没有sed_model，所以你的model实际上什么都没有load（你设了strick=False，它忽略所有的没有找到的weight，这里应该设为true的

解决方法很简单，把前缀裁掉就可以了，看这个就好了，我把你代码修改了一下，我随便试了10个结果，和答案都是一致的
[test_esc.zip](https://github.com/RetroCirce/HTS-Audio-Transformer/files/9147743/test_esc.zip)

**Comment by RetroCirce** [comment]
顺便我把单卡gpu和esc的config也更新了，你可以看看我最新的commit。

**Comment by joewale** [comment]
问题解决了，谢谢哈！
另外，有个问题，请教下，模型输出的pred_prob的概率得分的区间是怎样的，6.831792代表的意义是什么？

**Comment by RetroCirce** [comment]
你可以在最后softmax一下，就是概率了，因为esc用的是cross_entropy而不是BCE，默认的torch loss函数是自动帮你softmax了（又或者你可以手动做nllloss），所以我输出的时候就没有softmax

**Comment by joewale** [comment]
谢谢你的解答，如你所说，加上softmax就可以了，再次感谢！
`             
                pred = output_dict['clipwise_output']
  
                pred = F.softmax(pred, dim=1)

                pred_post = pred[0].detach().cpu().numpy()

                pred_label = np.argmax(pred_post)

                pred_prob = round(np.max(pred_post)*100, 3)`
